Orange/canvas/application/addons.py

```diff
@@ -760,127 +770,156 @@ def setStatusMessage(self, message):
-                                     universal_newlines=True,
-                                     env=_env_with_proxies()
-                                    )
-            retcode, output = self.__subprocessrun(process)
-
-            if retcode != 0:
-                self.error.emit(fmt_cmd(cmd), pkg, retcode, output)
-                return
-
-            # Why is this here twice??
-            cmd = (["-m", "pip", "install"] +
-                   (["--user"] if self.__user_install else []) +
-                   [inst_name])
-            process = python_process(cmd,
-                                     bufsize=-1,
-                                     universal_newlines=True,
-                                     env=_env_with_proxies()
-                                    )
-            retcode, output = self.__subprocessrun(process)
-
-            if retcode != 0:
-                self.error.emit(fmt_cmd(cmd), pkg, retcode, output)
-                return
-
-        elif command == Uninstall:
-            dist = pkg.local
-            self.setStatusMessage("Uninstalling {}".format(dist.project_name))
-
-            cmd = ["-m", "pip", "uninstall", "--yes", dist.project_name]
-            process = python_process(cmd,
-                                     bufsize=-1,
-                                     universal_newlines=True,
-                                     env=_env_with_proxies()
-                                    )
-            retcode, output = self.__subprocessrun(process)
-
-            if self.__user_install:
-                # Remove the package forcefully; pip doesn't (yet) uninstall
-                # --user packages (or any package outside sys.prefix?)
-                # google: pip "Not uninstalling ?" "outside environment"
-                install_path = os.path.join(
-                    USER_SITE, re.sub('[^\w]', '_', dist.project_name))
-                pip_record = next(iglob(install_path + '*.dist-info/RECORD'),
-                                  None)
-                if pip_record:
-                    with open(pip_record) as f:
-                        files = [line.rsplit(',', 2)[0] for line in f]
-                else:
-                    files = [os.path.join(
-                        USER_SITE, 'orangecontrib',
-                        dist.project_name.split('-')[-1].lower()),]
-                for match in itertools.chain(files, iglob(install_path + '*')):
-                    print('rm -rf', match)
-                    if os.path.isdir(match):
-                        shutil.rmtree(match)
-                    elif os.path.exists(match):
-                        os.unlink(match)
-
-            if retcode != 0:
-                self.error.emit(fmt_cmd(cmd), pkg, retcode, output)
-                return
+        try:
+            if command == Install:
+                self.setStatusMessage(
+                    "Installing {}".format(pkg.installable.name))
+                if self.conda:
```

Comment:
If `self.conda` is an instance of `CondaInstaller()`, shouldn't this check be `self.conda.conda`?

Reply:
Technically yes, but since CondaInstaller [reimplements `__bool__`](https://github.com/biolab/orange3/pull/2561/files#diff-55a6a7f617395a75195203c03b442985R881), this is the same thing